lib/routes/account.js

```diff
@@ -1612,4 +1630,50 @@ module.exports = function (
   }
 
   return routes
+
+  function upsertDevice (request, sessionToken, deviceInfo) {
+    var operation = deviceInfo.id ? 'updateDevice' : 'createDevice'
+    return db[operation](sessionToken.uid, sessionToken.tokenId, deviceInfo)
+      .then(
+        function (device) {
+          if (operation === 'createDevice') {
+            log.event('device:create', request, {
+              uid: sessionToken.uid,
+              id: device.id,
+              type: device.type,
+              timestamp: device.createdAt
+            })
+          }
+          push.notifyDeviceConnected(sessionToken.uid, device.name, device.id.toString('hex'))
```

Comment:
Is it correct for this to be called in both the `createDevice` and `updateDevice` cases? It was already before this change but the semantics of that seem funny to me, e.g. in the event that the device name has changed.

This change also means it would get called more frequently than it does already, whenever placeholder devices are updated to real ones.